### PR TITLE
DX to openGL/openGLES transpilation fixes

### PR DESCRIPTION
**Pull request by deltaoscarmike**
Improve GLES support: all presets with shaders are now successfully transpiled to openGL ES

It should fix one issue seen in #115

### CODE REVIEW

File: src/libprojectM/Renderer/ShaderEngine.cpp [hunk 1]
```diff
@@ -19,8 +19,10 @@
 
 #ifdef USE_GLES
     #define GLSL_VERSION "300 es"
+    #define SHADER_VERSION  M4::GLSLGenerator::Version_300_ES
```
Comment (revmischa) [inline src/libprojectM/Renderer/ShaderEngine.cpp]: i think this still won't fly on a raspi will it?

File: src/libprojectM/Renderer/hlslparser/src/GLSLGenerator.cpp [hunk 2]
```diff
@@ -1826,47 +1875,59 @@ void GLSLGenerator::OutputDeclaration(HLSLDeclaration* declaration)
...
+                globalVarsAssignements.push_back(declaration);
+            }
+        }
 
 		lastDecl = declaration;
 		declaration = declaration->nextDeclaration;
 	}
 }
 
+void GLSLGenerator::OutputDeclarationAssignement(HLSLDeclaration* declaration)
```
Comment (revmischa) [inline src/libprojectM/Renderer/hlslparser/src/GLSLGenerator.cpp]: Assignment, not Assignement

File: src/libprojectM/Renderer/ShaderEngine.cpp (hunk 1 above)
Comment (deltaoscarmike) [reply src/libprojectM/Renderer/ShaderEngine.cpp]: no, hopefully gles3 support will be implemented on broadcom vc4 driver someday.

File: src/libprojectM/Renderer/hlslparser/src/GLSLGenerator.cpp (hunk 2 above)
Comment (deltaoscarmike) [reply src/libprojectM/Renderer/hlslparser/src/GLSLGenerator.cpp]: done

**Comment by deltaoscarmike** [comment]
GL mode (desktop GL or GLES) should be detected on app launch and not rely on a compilation flag.

**Pull request merged**